### PR TITLE
Update developer/maintainer docs for release

### LINKED ISSUE #65: Clean and update dev docs
The dev docs should be and cleaned.

- [x] Fix FIXMEs
- [x] Fix TODOs
- [x] Check that docs are presentable

Optional: Update with relevant info (i.e., the small things that we are using but aren't documented yet, like the intricacies of Tycho, etc.). If not finished by milestone, move to new issue.

### LINKED ISSUE #73: Improve bugfix/hotfix workflow
We may want to differentiate between bug fixes from external users and hotfixes by maintainers.

Hotfixes would follow the GitFlow hotfix workflow.

Bug fixes could just be features, and released as/collected in a feature release.

If we wanted to stick as close as possible to GitFlow, we could ask people to use any old branch for a bug fix, then start the hotfix workflow locally, and merge their branch into the hotfix, solving conflicts ourselves (as the maintainer), etc. This would IMHO be the better option, separate bug fixes, and automate SemVer compatibility a bit better.

Or, they just submit a PR as a feature, and we simply follow the feature release workflow. In which case using `mvn gitflow:feature-start` would have pulled in the wrong version bump, i.e., in minor rather than patch.

Perhaps we could just have both scenarios.

Suggestion: Set this up in a fork, try all options, then decide.

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [X] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)

## Pull request type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] Documentation content changes
- [X] Other (please describe): 

- GitHub templates changed (PR template)

## What is the current behavior?

Developer/maintainer docs aren't complete and include TODOs and FIXMEs

Issue Number: #65 

## What is the new behavior?

- Add dependencies and templates checks to PR template
- Update developer/maintainer documentation (#65)
-

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

Fixes #65.
Fixes #73.

### Related issues not addressed in this PR

Kept for future PRs:
- #58 
- #61 
- #48 
- #45 
- #32 
- #41
- #84
- #85
- #86
- #87

---

# Checklist for maintainers

## Releases

- [ ] Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.

### CODE REVIEW

**Comment by sdruskat** [comment]
@thomaskrause At review, please make sure to review https://github.com/hexatomic/hexatomic/pull/83/commits/7c8e9553cd2502313fa3c87e884741689163eaa1 carefully for correctness :)

**Comment by sdruskat** [comment]
Made sure this PR gets the dev docs into decent shape. More work needs to be done on the dev docs, which I've documented in new issues (#84, #85, #86, #87).

File: docs/dev/src/development/workflow/repositories.md [hunk 1]
```diff
@@ -1 +1,68 @@
...
+The target platform for Hexatomic is defined using the []() plug-in for Eclipse.
+This allows to define the platform in a human-readable format, in the file [`org.corpus_tools.hexatomic.target.tpd`](https://github.com/hexatomic/hexatomic/blob/develop/releng/org.corpus_tools.hexatomic.target/org.corpus_tools.hexatomic.target.tpd).
+The actual [target platform definition file](https://github.com/hexatomic/hexatomic/blob/develop/releng/org.corpus_tools.hexatomic.target/org.corpus_tools.hexatomic.target.target) that both the Eclipse IDE and the Tycho Maven Plugin (which is used to build Hexatomic) use is generated
+from this file.
+More details about target platforms can be found in the section [*Creating and activating the target platform*](../activating-target-platform.html).
+
+#### Documentation
+
+Both the user documentation and the documentation for developers and maintainers are kept in the [`/docs/` folder](https://github.com/hexatomic/hexatomic/tree/develop/docs), in a subfolder for each documentation type.
+Documentation is written in Markdown from which websites are generated for the end user with *mdbook* (see section [*Development setup*](setup.html#mdbook)).
```
Comment (thomaskrause) [inline docs/dev/src/development/workflow/repositories.md]: The link to the development setup page is wrong and I'm **asking to change** it.
```suggestion
Documentation is written in Markdown from which websites are generated for the end user with *mdbook* (see section [*Development setup*](../setup.html#mdbook)).
```

File: docs/dev/src/development/workflow/README.md [hunk 2]
```diff
@@ -62,36 +72,38 @@ git fetch
...
 ```
-2. Create the new feature branch:  
+3. Create the new feature branch:  
 ```bash
 mvn gitflow:feature-start
 ```
-3. Give the feature branch a name as prompted by the Maven Gitflow plugin.
+4. Give the feature branch a name as prompted by the Maven Gitflow plugin.
 4. [Create a draft pull request](#create-a-pull-request-for-your-contribution-before-you-start-working) against the `develop` branch of <https://github.com/hexatomic/hexatomic> and start working.
+4. Make, commit, and push your changes to this branch.
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply docs/dev/src/development/workflow/README.md]: Fixed in https://github.com/hexatomic/hexatomic/pull/83/commits/fb742399afb9899a92284cd3c7e2c9451f9e1f2f, https://github.com/hexatomic/hexatomic/pull/83/commits/0b384a0bd7d53f773207dbe66a8bf014507a8300.

**Pull request merged**